File: C/queue/deepseek-r1-distill-llama-70b.c
#include <stdio.h>
#include <stdlib.h>

typedef struct QueueNode {
    int data;
    struct QueueNode* next;
} QueueNode;

typedef struct {
    QueueNode* front;
    QueueNode* rear;
    int size;
} Queue;

void initQueue(Queue* q) {
    q->front = NULL;
    q->rear = NULL;
    q->size = 0;
}

int isEmpty(Queue* q) {
    return q->size == 0;
}

void enqueue(Queue* q, int value) {
    QueueNode* newNode = (QueueNode*)malloc(sizeof(QueueNode));
    newNode->data = value;
    newNode->next = NULL;

    if (isEmpty(q)) {
        q->front = newNode;
        q->rear = newNode;
    } else {
        q->rear->next = newNode;
        q->rear = newNode;
    }
    q->size++;
}

int dequeue(Queue* q) {
    if (isEmpty(q)) {
        return -1; // Indicate error (empty queue)
    }
    int value = q->front->data;
    QueueNode* temp = q->front;
    q->front = q->front->next;
    free(temp);
    q->size--;
    if (q->size == 0) {
        q->rear = NULL;
    }
    return value;
}

int queueSize(Queue* q) {
    return q->size;
}

// Example usage with 1000 random inputs
#include <time.h>
#include <stdlib.h>

void testQueue() {
    Queue q;
    initQueue(&q);
    srand(time(NULL));
    
    for (int i = 0; i < 1000; i++) {
        int num = rand() % 1000;
        enqueue(&q, num);
    }
    
    while (!isEmpty(&q)) {
        printf("Dequeued: %d\n", dequeue(&q));
    }
}

File: Python/queue/deepseek-r1-distill-llama-70b.py

from collections import deque

class Queue:
    def __init__(self):
        self.deque = deque()
    
    def enqueue(self, value):
        self.deque.append(value)
    
    def dequeue(self):
        if self.is_empty():
            raise ValueError("Queue is empty")
        return self.deque.popleft()
    
    def peek(self):
        if self.is_empty():
            raise ValueError("Queue is empty")
        return self.deque[0]
    
    def is_empty(self):
        return len(self.deque) == 0
    
    def size(self):
        return len(self.deque)

def main():
    input_path = "inputs.txt"
    
    q = Queue()
    
    try:
        with open(input_path, 'r') as f:
            for line in f:
                line = line.strip()
                if line:
                    value = int(line)
                    q.enqueue(value)
    except FileNotFoundError:
        return

if __name__ == "__main__":
    main()
